### DISCUSSION TITLE
JavaFX 18

### DISCUSSION THREAD

**Discussion by georgweiss** [ControlSystemStudio/phoebus#2177] [General]
JavaFX 18 is now available from https://openjfx.io/ and Maven Central.

Tested OPIs, Data Browser, Alarm Tree and Olog on:

- Windows 10
- Mac OS Catalina (and basic test on Monterrey)
- CentOS 7 with xfce 

All using JRE 11.

No issues spotted. Created branch JavaFX_18 with minimal change in top level pom.

There is some urgency to update to 18 as Mac users are updating to Monterrey, see #2056.

**Comment by kasemir** [comment]
I suggest we go ahead.

I'm a little concerned about https://gluonhq.com/products/javafx/openjfx-18-release-notes/ deprecating `-Djdk.gtk.version=2`.
We have several issues that are currently "fixed" by using GTK2:  #367 #640 #1403
 But that's a deprecation warning, for now we can continue to use `-Djdk.gtk.version=2`, so again I'd go ahead.

**Reply by georgweiss** [reply]
Yes, I found that browsing through the release notes... The issues you mention, are they on Gnome? Maybe implied by *g*tk?

**Reply by kasemir** [reply]
Certainly all limited to Linux, X11, GTK. Not sure if the window manager/desktop matters

**Reply by georgweiss** [reply]
Cannot say my users have reported anything related to the mentioned issue tickets. But then again, we're on xfce...

**Reply by georgweiss** [reply]
Correction: #1403 has been seen, but ignored.

**Reply by kasemir** [reply]
GDM, XFCE, ..., there's also users connecting via ssh/XQuartz from mac, or cygwin from Windows. The lowest common denominator is likely whatever older mechanisms GTK2 uses. But we can't stay with that forever, will eventually have to drop the `-Djdk.gtk.version=2`

**Reply by georgweiss** [reply]
Agree.

**Comment by shroffk** [comment]
> All using JRE 11.

As long as the above has been tested I think we should be able to move to openjfx18.
Considering we are already on openjfx16.. it might be good to just move toopenjfx18 which matches the new LTS

**Comment by georgweiss** [comment]
Fine, I'll create the PR.
